python/tk_desktop/rpc.py

```diff
@@ -278,10 +278,17 @@ def run(self):
                         raise
                     ready = False
             else:
-                # can use select on osx and linux
-                (rd, _, _) = select.select(
-                    [self.server._listener._socket], [], [], self.LISTEN_TIMEOUT
-                )
+                # Can use select on OSX and Linux.
+                try:
+                    (rd, _, _) = select.select(
+                        [self.server._listener._socket], [], [], self.LISTEN_TIMEOUT
+                    )
+                except select.error as e:
+                    if "Bad file descriptor" in e.args:
```

Comment:
Is there a better way of catching this?

Reply:
We could in theory wait a short while and check the `self._is_closed` variable before deciding to break or raise.
I noticed that sometimes if I printed the `self._is_closed` in the exception block it would be false and sometimes true. I presume this is because it failed due to it being in the process of closing, but hadn't actually finished so the flag wasn't set.